implementations/micrometer-registry-dynatrace/src/test/java/io/micrometer/dynatrace/DynatraceConfigTest.java

```diff
@@ -261,4 +279,42 @@ public DynatraceApiVersion apiVersion() {
 
         assertThat(config.apiVersion()).isEqualTo(DynatraceApiVersion.V1);
     }
+
+    @Test
+    void testFileBasedConfig() throws IOException, InterruptedException {
+        String uuid = UUID.randomUUID().toString();
+        final Path tempFile = Files.createTempFile(uuid, ".properties");
+
+        Files.write(tempFile,
+                ("DT_METRICS_INGEST_URL = https://your-dynatrace-ingest-url/api/v2/metrics/ingest\n" +
+                        "DT_METRICS_INGEST_API_TOKEN = YOUR.DYNATRACE.TOKEN").getBytes());
+        // wait for write operation to finish, as it is running async.
```

Comment:
Can you clarify this: Are you waiting for the file to be written? Or are you waiting so the poller has picked it up? Isn't this write operation blocking? https://stackoverflow.com/questions/55882261/is-java-nio-file-files-createfile-a-blocking-call? I'm asking because of the sleep below.. I feel it might be flaky so if the file takes a bit longer to write then the test will fail.

Reply:
You are right, I thought NIO stands for non-blocking IO. It doesn't. It stands for new IO, which, incidentally, also has support for non-blocking IO (probably why i got confused). I updated the comment.

Reply:
It seems to still require the timeout, without it it sometimes fails. I think its due to the fact that the same thread writes and reads the file in such quick succession, that the FilePoller does not update in time. I think this is only a problem for tests, as in "the real world" the file will be written by a different process, and those events should never happen so close after one another and even if they did it would not matter. The next export would then use the updated token anyway.